src/System.Net.HttpListener/src/System/Net/Managed/WebSockets/HttpWebSocket.Managed.cs

```diff
@@ -75,45 +77,7 @@ internal static partial class HttpWebSocket
             return webSocketContext;
         }
 
-        private static void ValidateWebSocketHeaders(HttpListenerContext context)
-        {
-            if (!context.Request.IsWebSocketRequest)
-            {
-                throw new WebSocketException(WebSocketError.NotAWebSocket,
-                    SR.Format(SR.net_WebSockets_AcceptNotAWebSocket,
-                    nameof(ValidateWebSocketHeaders),
-                    HttpKnownHeaderNames.Connection,
-                    HttpKnownHeaderNames.Upgrade,
-                    HttpWebSocket.WebSocketUpgradeToken,
-                    context.Request.Headers[HttpKnownHeaderNames.Upgrade]));
-            }
-
-            string secWebSocketVersion = context.Request.Headers[HttpKnownHeaderNames.SecWebSocketVersion];
-            if (string.IsNullOrEmpty(secWebSocketVersion))
-            {
-                throw new WebSocketException(WebSocketError.HeaderError,
-                    SR.Format(SR.net_WebSockets_AcceptHeaderNotFound,
-                    nameof(ValidateWebSocketHeaders),
-                    HttpKnownHeaderNames.SecWebSocketVersion));
-            }
-
-            if (!string.Equals(secWebSocketVersion, SupportedVersion))
-            {
-                throw new WebSocketException(WebSocketError.UnsupportedVersion,
-                    SR.Format(SR.net_WebSockets_AcceptUnsupportedWebSocketVersion,
-                    nameof(ValidateWebSocketHeaders),
-                    secWebSocketVersion,
-                    SupportedVersion));
-            }
-
-            if (string.IsNullOrWhiteSpace(context.Request.Headers[HttpKnownHeaderNames.SecWebSocketKey]))
-            {
-                throw new WebSocketException(WebSocketError.HeaderError,
-                    SR.Format(SR.net_WebSockets_AcceptHeaderNotFound,
-                    nameof(ValidateWebSocketHeaders),
-                    HttpKnownHeaderNames.SecWebSocketKey));
-            }
-        }
+        private static void ValidateWebSocketHeadersCore(HttpListenerContext context) { }
```

Comment:
You could also do this as a partial method, in which case you won't need an empty method like this in the managed implementation.